src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/CommunityOffenderManager.kt

```diff
@@ -1,33 +1,62 @@
 package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity
 
-import jakarta.persistence.Column
+import jakarta.persistence.DiscriminatorValue
 import jakarta.persistence.Entity
-import jakarta.persistence.GeneratedValue
-import jakarta.persistence.GenerationType
-import jakarta.persistence.Id
-import jakarta.persistence.Table
-import jakarta.validation.constraints.NotNull
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.StaffKind
 import java.time.LocalDateTime
 
 @Entity
-@Table(name = "community_offender_manager")
-data class CommunityOffenderManager(
-  @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  @NotNull
-  val id: Long = -1,
-
-  @Column(unique = true)
+@DiscriminatorValue(value = "COMMUNITY_OFFENDER_MANAGER")
+class CommunityOffenderManager(
+  id: Long? = -1,
   val staffIdentifier: Long,
-
-  @Column(unique = true)
-  override val username: String,
-
-  val email: String?,
-
-  override val firstName: String?,
-
-  override val lastName: String?,
-
-  val lastUpdatedTimestamp: LocalDateTime? = null,
-) : Creator
+  username: String,
+  email: String?,
+  firstName: String?,
+  lastName: String?,
+  lastUpdatedTimestamp: LocalDateTime? = null,
+) : Creator, Staff(
+  id = id,
+  kind = StaffKind.COMMUNITY_OFFENDER_MANAGER,
+  username = username,
+  email = email,
+  firstName = firstName,
+  lastName = lastName,
+  lastUpdatedTimestamp = lastUpdatedTimestamp,
+) {
+  fun copy(
```

Comment:
not sure if we want to consider the Lombok's Builder with copy constructors and use of With might reduce our boilerplate code

Reply:
There is a lot of boliler plate when not using data classes!

[lombok/kotlin](https://kotlinlang.org/docs/lombok.html) support is a bit experimental at the moment

I think we should probably embrace mutability a bit in the JPA code